pydantic/class_validators.py

```diff
@@ -244,86 +244,50 @@ def make_generic_validator(validator: AnyCallable) -> 'ValidatorCallable':
             f'Invalid signature for validator {validator}: {sig}, "self" not permitted as first argument, '
             f'should be: (cls, value, values, config, field), "values", "config" and "field" are all optional.'
         )
-    elif first_arg == 'cls':
-        # assume the second argument is value
-        return wraps(validator)(_generic_validator_cls(validator, sig, set(args[1:])))
     else:
-        # assume the first argument was value which has already been removed
-        return wraps(validator)(_generic_validator_basic(validator, sig, set(args)))
+        if first_arg == 'cls':
+            # assume the second argument is value
+            args = args[1:]
+            skip_first = False
+        else:
+            # assume the first argument was value which has already been removed
+            skip_first = True
+
+        return wraps(validator)(_generic_validator(validator, sig, set(args), skip_first))
```

Comment:
just noticed that I think the wraps here is redundant now that it's in the _generic_validator function

Reply:
Totally agree with you, thanks.